lib/resque/plugins/timed_round_robin/timed_round_robin.rb

```diff
@@ -6,13 +6,15 @@ def filter_busy_queues(qs)
     end
 
     def rotated_queues
-      # Grab the current list of queues.  Don't cache it beyond this method call
-      # because queues can be added/removed dynamically
-      @rtrr_queues = queues
+      # only refresh queues when a slice expires
+      @rtrr_queues ||= queues
       return [] if @rtrr_queues.empty?
 
       @n ||= 0
-      advance_offset if slice_expired?
+      if slice_expired?
+        advance_offset
+        @rtrr_queues = queues
```

Comment:
I think you want to flip these two lines, otherwise `advance_offset` is operating on the old queue list length.